lib/modules/download/requesters/search.rb

```diff
@@ -16,19 +16,19 @@ def domain
     'search'
   end
 
-  private
+
 
   def identifier
     token
   end
 
   def token
     @token ||= begin
-      filters_dump = Marshal.dump filters.keys.sort
+      filters_dump = Marshal.dump filters.sort.to_json
       Digest::SHA256.hexdigest(@search_term.to_s + filters_dump)
     end
   end
-
+  private
```

Comment:
it might be good to add a space before and after private to improve readability.